CRM/Utils/System/Standalone.php

```diff
@@ -0,0 +1,628 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+/**
+ * Standalone specific stuff goes here.
+ */
+class CRM_Utils_System_Standalone extends CRM_Utils_System_Base {
+
```

Comment:
I see you've removed registerPathVars() in this latest update. It means that `[cms.root]` is now null, which kind of makes sense here, but we'll need to hunt down where it's used since it generates errors. Or perhaps it should point to `web`.

Reply:
hmm I have to admit I don't really recall.  I grepped a bit in the codebase about `cms.root` and it seems there more for convenience, for settings, but I'd be surprised if extensions used it? (there are better functions for generating URLs and getPaths)

Reply:
Also I was using php 8 this time around, which is less forgiving. I didn't intend to, just it happened to be in my path from something else and then composer(👿) locked everything to php 8+ so it won't even run with a web server using 7.